src/lib/containers/entity/page/title_bar/EntityPageTitleBar.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react'
+import IconSvg, { type2SvgIconName } from '../../../IconSvg'
+import { Avatar, Box, Skeleton, Stack, Typography } from '@mui/material'
+import { useGetEntityBundle } from '../../../../utils/hooks/SynapseAPI'
+import FavoriteButton from '../../../favorites/FavoriteButton'
+import ConditionalWrapper from '../../../utils/ConditionalWrapper'
+import EntityActionMenu, {
+  EntityActionMenuProps,
+} from '../action_menu/EntityActionMenu'
+import { isDockerRepository } from '../../../../utils/synapseTypes/docker/DockerRepository'
+import TitleBarProperties from './TitleBarProperties'
+import { EntityTitleBarVersionInfo } from './EntityTitleBarVersionInfo'
+
+export type EntityPageTitleBarProps = {
+  entityId: string
+  versionNumber?: number
+  /* Props that will be passed through to an EntityActionMenu component */
+  entityActionMenuProps?: EntityActionMenuProps
+  /* Callback invoked when an ACT member clicks "Add Conditions for Use". */
+  onActMemberClickAddConditionsForUse: () => void
+}
+
+/**
+ * The EntityPageTitleBar component is shown on every Entity page in Synapse, and shows an Entity's metadata and
+ * provides controls to view and manipulate the Entity.
+ */
+export default function EntityPageTitleBar(props: EntityPageTitleBarProps) {
+  const { entityId, versionNumber, entityActionMenuProps } = props
+
+  const toggleShowVersionHistory =
+    entityActionMenuProps?.actionConfiguration['SHOW_VERSION_HISTORY']?.onClick
```

Comment:
Pulling this handler out of the action menu props is way easier than setting up a separate prop and accepting the same handler in two different places.